src/explore-education-statistics-admin/src/pages/release/content/components/__tests__/EditableKeyStatText.test.tsx

```diff
@@ -167,4 +171,160 @@ describe('EditableKeyStatText', () => {
       screen.queryByRole('button', { name: /Remove/ }),
     ).not.toBeInTheDocument();
   });
+
+  describe('when editing', () => {
+    test('renders edit form correctly', async () => {
+      render(
+        <EditableKeyStatText keyStat={keyStatText} isEditing onSubmit={noop} />,
+      );
+
+      await waitFor(() => {
```

Comment:
Generally we try to not have a lot in the waitFors and to especially avoid getByRole as this causes slowness running the tests. You can just wait for one thing then continue to check the rest after it.

Reply:
The `ByText`, `ByLabel` or `ByTestId` selectors are much much faster and are acceptable to use in the `waitFor`